src/main/java/com/revature/services/QuestionLogicImpl.java

```diff
@@ -32,12 +57,79 @@
 	@Override
 	public Page<QuestionPool> searchQuestions(Pageable pageable, String text, String subject) {
 		
-		if(subject.equalsIgnoreCase("all")){
+		if("all".equalsIgnoreCase(subject)){
 			return dao.findBySearchTerm(pageable, text);
 		} else {
 			return dao.findBySearchTerm(pageable, text, subject);
 		}
 
 	}
 	
+	@Override
+	public QuestionPool getQuestionById(Integer id) {
+        QuestionPool question = dao.findOne(id);
+
+        if (question == null) {
+            throw new NotFoundException("Question with id " + id + " not found");
+        }
+
+		return question;
+	}
+
+//UPDATE-----------------------------------
+	@Override
+	@Transactional
+	public QuestionPool updateQuestion(QuestionPool question, Integer questionId) {
+        
+        validation.validateQuestionPoolExists(questionId);
+
+        QuestionPool currQuestion = dao.findOne(questionId);
+
+		if(question.getQuestionText() != null){
+			currQuestion.setQuestionText(question.getQuestionText());
+		}
+		if(question.getMaxCommunicationScore() != null){
+            // TODO Add logic to find questions that use this and scale scores based on new maximum
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1135)